core/dbt/include/global_project/macros/materializations/models/table/columns_spec_ddl.sql

```diff
@@ -37,13 +37,34 @@
   {#--Obtain the column schema provided by the schema file by generating an 'empty schema' query from the model's columns. #}
   {%- set schema_file_provided_columns = get_column_schema_from_query(get_empty_schema_sql(model['columns'])) -%}
 
+  {#-- Build compiler error msg #}
   {%- set sql_file_provided_columns_formatted = format_columns(sql_file_provided_columns)  -%}
   {%- set schema_file_provided_columns_formatted = format_columns(schema_file_provided_columns)  -%}
+  {%- set compiler_error_msg = 'Please ensure the name, data_type, and number of columns in your `yml` file match the columns in your SQL file.\nSchema File Columns: ' ~ (schema_file_provided_columns_formatted|trim) ~ '\n\nSQL File Columns: ' ~ (sql_file_provided_columns_formatted|trim) ~ ' ' -%}
 
-  {%- if sql_file_provided_columns_formatted != schema_file_provided_columns_formatted -%}
-    {%- do exceptions.raise_compiler_error('Please ensure the name, data_type, order, and number of columns in your `yml` file match the columns in your SQL file.\nSchema File Columns: ' ~ (schema_file_provided_columns_formatted|trim) ~ '\n\nSQL File Columns: ' ~ (sql_file_provided_columns_formatted|trim) ~ ' ' ) %}
+
+  {%- if sql_file_provided_columns|length != schema_file_provided_columns|length -%}
+    {%- do exceptions.raise_compiler_error(compiler_error_msg) -%}
```

Comment:
Should we be raising a unique (new) error here?  I know it was already raising a CompilerError but it was added for multi project.  In the view materialization changes, I created a new error for it.

Reply:
What was the new error?

Reply:
These are the two new pieces I had to add.

https://github.com/dbt-labs/dbt-core/blob/f24452a3ab789d8e678f9833184286dfaf0b16ed/core/dbt/context/exceptions_jinja.py#L96-L97

https://github.com/dbt-labs/dbt-core/blob/f24452a3ab789d8e678f9833184286dfaf0b16ed/core/dbt/exceptions.py#L2095-L2110

You would need to add a new one.  I started a slack thread about larger picture how we wanted to handle these types of errors generated in jinja.

Reply:
I added "rasie_contract_error", ContractError.